### PR TITLE
Remain only InputModelTF constructor with GraphIterator and adopt other code

### PR DESCRIPTION
**Pull request by rkazants**
**Details**: 
- InputModelTF construction happens with GraphIterator only
- Different GraphIterator are responsible for different TensorFlow formats. Now we have only GraphIteratorProto responsible for binary protobuf format
- FrontEnd load method prepares GraphIterator for a given format
- all truly private methods (get_op_nodes, etc.) of InputModelTF are moved to private section after proper code-refactoring

**Ticket**: 62697

### CODE REVIEW

File: ngraph/frontend/tensorflow/include/tensorflow_frontend/frontend.hpp [hunk 1]
```diff
@@ -5,18 +5,11 @@
...
-using namespace ngraph::frontend;
-
-namespace tensorflow {
-class GraphDef;
-class NodeDef;
-namespace ngraph_bridge {
-class GraphIteratorProto;
-}
-}  // namespace tensorflow
+using namespace ::ngraph::frontend;
```
Comment (itikhono) [inline ngraph/frontend/tensorflow/include/tensorflow_frontend/frontend.hpp:12]: "using" in the header file?
not sure if this is ok, may be the cause of the name collision

File: ngraph/frontend/tensorflow/include/tensorflow_frontend/model.hpp [hunk 2]
```diff
@@ -19,17 +20,12 @@ class TF_API InputModelTF : public InputModel {
     class InputModelTFImpl;
     std::shared_ptr<InputModelTFImpl> _impl;
 
-public:
-    // TODO: move to private once GraphTranslation will be a part of FrontEndTF component
     std::vector<std::shared_ptr<OpPlaceTF>> get_op_places() const;
```
Comment (itikhono) [inline ngraph/frontend/tensorflow/include/tensorflow_frontend/model.hpp:23]: now we can get this information from GraphTranslation, right?

Comment (rkazants) [reply ngraph/frontend/tensorflow/include/tensorflow_frontend/model.hpp:23]: yep

File: ngraph/frontend/tensorflow/src/frontend.cpp [hunk 3]
```diff
@@ -12,20 +12,45 @@
...
+
+    NodeContext node_ctx(*node->get_decoder(), named_inputs);
+    auto new_node_outputs = translator_it->second(node_ctx);
+    SetTracingInfo(node_ctx.get_name(), new_node_outputs.front());
+
+    auto new_output = new_node_outputs.begin();
+    auto old_outputs = node->outputs();
+    auto old_output = old_outputs.begin();
+
+    for (; new_output != new_node_outputs.end() && old_output != old_outputs.end(); ++old_output, ++new_output) {
```
Comment (itikhono) [inline ngraph/frontend/tensorflow/src/frontend.cpp:43]: we expect that `new_node_outputs.size() == old_outputs.size()`?
if so, it's better to add check here

File: ngraph/frontend/tensorflow/include/tensorflow_frontend/frontend.hpp [hunk 4]
```diff
@@ -61,8 +54,14 @@ class TF_API FrontEndTF : public FrontEnd {
     bool supported_impl(const std::vector<std::shared_ptr<Variant>>& variants) const override;
 
     InputModel::Ptr load_impl(const std::vector<std::shared_ptr<Variant>>& variants) const override;
+
+private:
+    static void translate_graph(const std::shared_ptr<InputModelTF>& model,
```
Comment (itikhono) [inline ngraph/frontend/tensorflow/include/tensorflow_frontend/frontend.hpp:59]: const std::string -> const std::string&

File: ngraph/frontend/tensorflow/src/frontend.cpp [hunk 5]
```diff
@@ -12,20 +12,45 @@
...
+        old_output->replace(*new_output);
+    }
+}
+}  // namespace
+
+void FrontEndTF::translate_graph(const std::shared_ptr<InputModelTF>& model,
+                                 const std::string model_name,
```
Comment (itikhono) [inline ngraph/frontend/tensorflow/src/frontend.cpp:50]: const std::string -> const std::string&

File: ngraph/frontend/tensorflow/src/frontend.cpp [hunk 6]
```diff
@@ -12,20 +12,45 @@
...
-
-using namespace ngraph::frontend;
-using namespace ngraph::frontend::tf;
+using namespace ::ngraph::frontend;
+using namespace ::ngraph::frontend::tf;
 using namespace ::tensorflow::ngraph_bridge;
 
-using ::tensorflow::GraphDef;
-
 namespace {
```
Comment (itikhono) [inline ngraph/frontend/tensorflow/src/frontend.cpp:19]: unnamed namespace?

File: ngraph/frontend/tensorflow/src/frontend.cpp [hunk 7]
```diff
@@ -12,20 +12,45 @@
...
+    auto type = node->get_op_type();
+
+    const auto TRANSLATE_OP_MAP = get_supported_ops();
+    auto translator_it = TRANSLATE_OP_MAP.find(type);
+    FRONT_END_OP_CONVERSION_CHECK(translator_it != TRANSLATE_OP_MAP.end(), "No translator found for ", type, " node.");
+
+    ngraph::OutputVector ng_inputs;
+    NamedInputs named_inputs;
+    size_t input_port_idx = 0;
+    for (auto& input : node->inputs()) {
```
Comment (itikhono) [inline ngraph/frontend/tensorflow/src/frontend.cpp:30]: auto& input -> const auto& input

File: ngraph/frontend/tensorflow/src/frontend.cpp (hunk 3 above)
Comment (mvafin) [reply ngraph/frontend/tensorflow/src/frontend.cpp:43]: We expect that new_node_output.size() >= old_outputs.size(). I think.

**Review by mvafin — APPROVED** [review]
LGTM

File: ngraph/frontend/tensorflow/src/frontend.cpp (hunk 6 above)
Comment (rkazants) [reply ngraph/frontend/tensorflow/src/frontend.cpp:19]: it means static function. it is used nowhere else.

File: ngraph/frontend/tensorflow/src/frontend.cpp (hunk 3 above)
Comment (rkazants) [reply ngraph/frontend/tensorflow/src/frontend.cpp:43]: I left this logic unchanged. I think Maxim to be aware of it better.

File: ngraph/frontend/tensorflow/include/tensorflow_frontend/frontend.hpp (hunk 1 above)
Comment (rkazants) [reply ngraph/frontend/tensorflow/include/tensorflow_frontend/frontend.hpp]: done.

File: ngraph/frontend/tensorflow/include/tensorflow_frontend/frontend.hpp (hunk 4 above)
Comment (rkazants) [reply ngraph/frontend/tensorflow/include/tensorflow_frontend/frontend.hpp:57]: fixed

File: ngraph/frontend/tensorflow/src/frontend.cpp (hunk 5 above)
Comment (rkazants) [reply ngraph/frontend/tensorflow/src/frontend.cpp]: fixed

File: ngraph/frontend/tensorflow/src/frontend.cpp (hunk 7 above)
Comment (rkazants) [reply ngraph/frontend/tensorflow/src/frontend.cpp]: fixed

**Pull request merged**